packages/react-examples/src/react-button/MenuButton/MenuButton.stories.tsx

```diff
@@ -0,0 +1,106 @@
+import * as React from 'react';
+import { ContextualMenu, IContextualMenuProps } from '@fluentui/react';
+import { MenuButton, MenuButtonProps } from '@fluentui/react-button';
+import { Menu, MenuItem, MenuList, MenuProps, MenuTrigger } from '@fluentui/react-menu';
+import { useBoolean } from '@fluentui/react-utilities';
+import { buttonBaseProps } from '../Button/Button.stories';
+import { Playground } from '../Playground';
+import { PlaygroundProps, PropDefinition } from '../Playground.types';
+
+type ExampleProps = { menuType?: string };
+
+const contextualMenuProps: IContextualMenuProps = {
+  items: [
+    {
+      key: 'a',
+      name: 'Item a',
+    },
+    {
+      key: 'b',
+      name: 'Item b',
+    },
+  ],
+};
+
+const MenuButtonWithConvergedMenu = (props: MenuButtonProps): JSX.Element => {
+  const { expanded, onMenuDismiss } = props;
+
+  const onOpenChange: MenuProps['onOpenChange'] = (ev, data) => {
+    if (!data.open) {
+      onMenuDismiss?.();
+    }
+  };
+
+  return (
+    <Menu open={expanded} onOpenChange={onOpenChange}>
+      <MenuTrigger>
+        <MenuButton {...props} />
+      </MenuTrigger>
+
+      <MenuList>
+        <MenuItem>Item a</MenuItem>
+        <MenuItem>Item b</MenuItem>
+      </MenuList>
+    </Menu>
+  );
+};
+
+const ExampleMenu = (props: MenuButtonProps & ExampleProps): JSX.Element => {
+  const { menuType, ...rest } = props;
+  return menuType === 'ContextualMenu' ? (
+    <MenuButton menu={<ContextualMenu {...contextualMenuProps} />} {...rest} />
```

Comment:
it seems that we're only adding this `menu` slot for compat with `ContextualMenu`. Can we remove this slot in favour of a migration guide similar to what we did in `Flex` and give an example of controlling the `ContextualMenu` with the new `MenuButton` as `target` ?

This would keep converged component codebase simpler without workarounds for legacy compatware and give partners a clear migration path to using the converged menu. The use of the `slot` might actually make migration from `ContextualMenu` and converged `Menu` harder because the API is just vastly different

Reply:
Resolving as per our offline conversation of integrating our efforts and having a stricter contract with `MenuTrigger`